src/dashboard/src/components/administration/forms.py

```diff
@@ -129,10 +139,306 @@ class AtomDipUploadSettingsForm(SettingsForm):
+        """
+        self.fields['name'].initial = name
+        self.fields['name'].widget.attrs['readonly'] = 'readonly'
+        config_path = os.path.join(helpers.processing_config_path(), '{}ProcessingMCP.xml'.format(name))
+        with open(config_path, 'r') as f:
+            xml = f.read()
+            root = etree.fromstring(xml)
+            for choice in root.findall('.//preconfiguredChoice'):
+                applies_to = choice.find('appliesTo').text
+                go_to_chain = choice.find('goToChain').text
+                fprops = self.processing_fields.get(applies_to)
+                field = self.fields.get(applies_to)
+                if fprops is None or go_to_chain is None or field is None:
+                    continue
+                if fprops['type'] == 'days':
+                    field.initial = int(float(choice.find('delay').text)) / (24 * 60 * 60)
+                else:
+                    field.initial = go_to_chain
+
+    def save_config(self):
+        """
+        Encode the configuration to XML and write it to disk.
+        """
+        name = self.cleaned_data['name']
+        del self.cleaned_data['name']
+        config_path = os.path.join(helpers.processing_config_path(), '{}ProcessingMCP.xml'.format(name))
+        config = PreconfiguredChoices(config_path)
+        for choice_uuid, value in self.cleaned_data.iteritems():
+            fprops = self.processing_fields.get(choice_uuid)
+            if fprops is None or value is None:
+                continue
+            field = self.fields.get(choice_uuid)
+            if field is None:
+                continue
+            if isinstance(field, forms.ChoiceField):
+                if not value: # Ignore empty string!
+                    continue
+            if fprops['type'] == 'days':
+                if value == 0:
+                    continue
+                delay = str(float(value) * (24 * 60 * 60))
+                config.add_choice(choice_uuid, fprops['chain'], delay_duration=delay)
+            else:
+                config.add_choice(choice_uuid, value)
+        config.save()
+
+
+def get_storage_locations(purpose):
+    try:
+        dirs = storage_service.get_location(purpose=purpose)
+        if len(dirs) == 0:
+            raise Exception('Storage server improperly configured.')
+    except Exception:
+        dirs = []
+    return dirs
+
+
+class PreconfiguredChoices:
+    """
+    Encode processing configuration XML documents and optionally write to disk.
+    """
+    def __init__(self, config_path):
+        self.xml = etree.Element('processingMCP')
+        self.choices = etree.Element('preconfiguredChoices')
+        self.config_path = config_path
+
+    def add_choice(self, applies_to_text, go_to_chain_text, delay_duration=None):
+        choice = etree.Element('preconfiguredChoice')
+
+        applies_to = etree.Element('appliesTo')
+        applies_to.text = applies_to_text
```

Comment:
`etree.Element('foo').text = 'text'` works, though you may not think it's cleaner. :)